manifests/base.pp

```diff
@@ -80,12 +84,24 @@
         settings => $shorewall::merged_settings;
     }
   }
+
+  $path        = [
+    '/usr/local/sbin',
+    '/usr/local/bin',
+    '/usr/sbin',
+    '/usr/bin',
+    '/sbin',
+    '/bin',
+  ]
```

Comment:
You should not do that: https://www.puppetcookbook.com/posts/set-global-exec-path.html

Reply:
Yes, of course it is possible. But many modules set a default value for a path because it adds autonomy and versatility to the module.

For example:
https://github.com/voxpupuli/puppet-nginx/blob/master/manifests/service.pp

Imho it’s not bad practice.

Reply:
If you want autonomy then why not the full path?

But if you clutter all over your modules the same piece of code, something smells.

Reply:
>If you want autonomy then why not the full path?

The full path will not allow the command to be executed in different environments.
For example:
Ubuntu 20.04:
/usr/sbin/shorewall
Ubuntu 18.04:
/sbin/shorewall

>But if you clutter all over your modules the same piece of code, something smells.
It’s true. I do not fully understand why the function exec does not set the default path parameter.